app/js/views/app_view.js

```diff
@@ -2,20 +2,25 @@ define([
   'jquery',
   'backbone',
   'views/todoList_view',
-  'views/addTodo_view'
-], function ($, Backbone, TodoListView, AddTodoView) {
+  'views/addTodo_view',
+  'views/filter_view',
+  'model/filter'
+], function ($, Backbone, TodoListView, AddTodoView, FilterView, Filter) {
   return Backbone.View.extend({
     initialize: function () {
+      var filter = new Filter();
+
       this.addTodoView = new AddTodoView({collection: this.collection});
-      this.todoListView = new TodoListView({collection: this.collection});
+      this.todoListView = new TodoListView({collection: this.collection, listFilter: filter});
+      this.filterView = new FilterView({collection: this.collection, listFilter: filter});
```

Comment:
I see that the filter instance is shared between the list and filter view. Can we come up with a better way to avoid triggering the "display" and "hide" events through the models ?